e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,286 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/math"
+	transfertypes "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFiatTFIBCOut(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw, gaia, r, ibcPathName, eRep := nobleSpinUpIBC(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, gaia)
+	nobleWallet := w[0]
+	gaiaWallet := w[1]
+
+	mintAmount := int64(100)
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.Minter.KeyName(), "fiat-tokenfactory", "mint", nobleWallet.FormattedAddress(), fmt.Sprintf("%duusdc", mintAmount))
+	require.NoError(t, err, "error minting")
+
+	// noble -> gaia channel info
+	nobleToGaiaChannelInfo, err := r.GetChannels(ctx, eRep, noble.Config().ChainID)
+	require.NoError(t, err)
+	nobleToGaiaChannelID := nobleToGaiaChannelInfo[0].ChannelID
+	// gaia -> noble channel info
+	gaiaToNobleChannelInfo, err := r.GetChannels(ctx, eRep, gaia.Config().ChainID)
+	require.NoError(t, err)
+	gaiaToNobleChannelID := gaiaToNobleChannelInfo[0].ChannelID
+
+	amountToSend := math.NewInt(5)
+	transfer := ibc.WalletAmount{
+		Address: gaiaWallet.FormattedAddress(),
+		Denom:   denomMetadataUsdc.Base,
+		Amount:  amountToSend,
+	}
+
+	// ACTION: IBC send out a TF token while TF is paused
+	// EXPECTED: Request fails;
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ibc transfer noble -> gaia
+	_, err = noble.SendIBCTransfer(ctx, nobleToGaiaChannelID, nobleWallet.KeyName(), transfer, ibc.TransferOptions{})
+	require.ErrorContains(t, err, "the chain is paused")
+
+	// relay MsgRecvPacket & MsgAcknowledgement
+	require.NoError(t, r.Flush(ctx, eRep, ibcPathName, nobleToGaiaChannelID))
+
+	// uusdc IBC denom on gaia
+	srcDenomTrace := transfertypes.ParseDenomTrace(transfertypes.GetPrefixedDenom("transfer", gaiaToNobleChannelID, denomMetadataUsdc.Base))
+	dstIbcDenom := srcDenomTrace.IBCDenom()
```

Comment:
If possible, these lines might fit in better with the setup (around line 44) rather than as part of the first test case since they appear to be shared across all the cases.

Reply:
Updated:
[`5df6693` (#374)](https://github.com/noble-assets/noble/pull/374/commits/5df669319f57580b2c778c1e32e09eeee3babeb2)